rust/agama-lib/src/network/client.rs

```diff
@@ -86,10 +86,12 @@ impl<'a> NetworkClient<'a> {
             .await?;
         let name = device_proxy.name().await?;
         let device_type = device_proxy.type_().await?;
+        let state = DeviceState::try_from(device_proxy.state().await?).expect("Unknown state");
 
         Ok(Device {
             name,
-            type_: DeviceType::try_from(device_type).unwrap(),
+            type_: DeviceType::try_from(device_type).expect("Unknown type"),
+            state,
```

Comment:
Do we really need to update this struct? In any case, I would expect this struct to be removed when we drop the D-Bus code.

Reply:
Probably not, should be dropped as you commented.